src/core/project/qgsprojectutils.cpp

```diff
@@ -52,3 +53,18 @@ bool QgsProjectUtils::updateLayerPath( QgsProject *project, const QString &oldPa
   }
   return res;
 }
+
+bool QgsProjectUtils::layerIsContainedInGroupLayer( QgsProject *project, QgsMapLayer *layer )
+{
+  const QMap<QString, QgsMapLayer *> mapLayers( project->mapLayers() );
```

Comment:
@nyalldawson , just an idea (not part of this PR), it might be useful to eventually add a QgsProject::mapLayersByType( Qgis::QgsMapLayerType ) function. It'd be useful here (and likely in other places too).

Reply:
good catch -- there's already a method for that. I've pushed a commit to use it